### ISSUE TITLE
Add typescript definitions

### ISSUE THREAD

**Issue by yordis** [expo/vector-icons#38]
Self descriptive title 😄

**Comment by yordis** [comment]
I dont know that much about TS yet but this could help https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-native-vector-icons

**Comment by brentvatne** [comment]
sure, feel free to add to DefinitelyTyped

**Comment by yordis** [comment]
@brentvatne sorry what?

I don't know Typescript enough for do any of these task or know the internals of any package ... the only reason why I created the ticket .

**Comment by brentvatne** [comment]
@yordis - we add typescript definitions for our projects. typescript types are largely managed by the community in the DefinitelyTyped repo. you're welcome to add them to there but we won't be writing and maintaining them for expo libraries in the near-term -- perhaps in the future! you can create a feature request on https://expo.canny.io/feature-requests to ask for typescript definitions if you want!

**Comment by brentvatne** [comment]
you can read more about creating a definitelytyped interface at http://definitelytyped.org/guides/creating.html

**Comment by incleaf** [comment]
I've added [definitions](https://www.npmjs.com/package/@types/expo__vector-icons) into `DefinitelyTyped`.
You can now simply use: `npm install --dev @types/expo__vector-icons` 😀

**Comment by indigoviolet** [comment]
Given that `@expo/vector-icons` has its own typescript files (which are unfortunately less useful than the DefinitelyTyped ones, for example they don't have good proptypes), what's the right way of consuming the `@types/expo__vector-icons` package?

I currently have 

```
import { MaterialCommunityIcons, MaterialIcons } from '@expo/vector-icons';
import { MaterialIcons as MaterialIconsT } from 'expo__vector-icons';
```

Is this expected or is there a nicer way of doing this?

**Comment by georgiosApo** [comment]
> Given that `@expo/vector-icons` has its own typescript files (which are unfortunately less useful than the DefinitelyTyped ones, for example they don't have good proptypes), what's the right way of consuming the `@types/expo__vector-icons` package?
> 
> I currently have
> 
> ```
> import { MaterialCommunityIcons, MaterialIcons } from '@expo/vector-icons';
> import { MaterialIcons as MaterialIconsT } from 'expo__vector-icons';
> ```
> 
> Is this expected or is there a nicer way of doing this?

I'm also interested in how to use this properly. Can't seem to get the squigglies away from my code.  @brentvatne maybe you can ad some insight to this? Thanks in advance!

**Comment by brentvatne** [comment]
it's not clear to me that the typescript definitions here are less useful. as explained in https://blog.expo.io/expo-sdk-40-is-now-available-d4d73e67da33 - for example, we now include icon names in types as well.

if you'd like to improve typescript support in this repo please open an issue with an exact definition of the problem you're encountering or a pr to fix that problem :)

**Comment by brentvatne** [comment]
this pr should fix some of the remaining outstanding issues, i've released it in `@expo/vector-icons@12.0.3`
